### ISSUE TITLE
Cannot cast to generic type

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#1248]
**Describe the bug**

There seems to be a regression in the latest pyright 1.1.93. Cannot cast variable to generic type.

**To Reproduce**

See code example below

**Expected behavior**

Should be possible to cast anything to any type.

**Screenshots or Code**
```python
from typing import Any, Generic, TypeVar, cast

TSource = TypeVar("TSource")


class Test(Generic[TSource]):
    def test(self, value: Any):
        cast(TSource, value)
```

![Screenshot 2020-12-07 at 09 50 39](https://user-images.githubusercontent.com/849479/101329436-af64a280-3871-11eb-9be0-65646426f8c4.png)

**VS Code extension or command-line**
Vscode extension:

[Info  - 7:59:14 AM] Pyright language server 1.1.93 starting
[Info  - 7:59:14 AM] Server root directory: /Users/dbrattli/.vscode-insiders/extensions/ms-pyright.pyright-1.1.93/dist
Received pythonPath from Python extension: /usr/local/opt/python@3.9/bin/python3

**Additional context**

This problem does not exist with Pyright language server 1.1.92.

**Comment by erictraut** [comment]
I don't think this is a regression. I'm pretty sure this would have generated an error in previous versions as well. The issue is that the definition of `cast` in `typing.pyi` contains two overloads. The first accepts a `Type[_T]`, and the second accepts a `str`. `TSource` doesn't match either of these. You can work around the issue by quoting `TSource`. Type checkers allow type annotations to be quoted, and at runtime `cast` doesn't do anything with the first argument, so its value doesn't matter at that time.
